stm32-modules/include/flex-stacker/flex-stacker/tmf8821.hpp

```diff
@@ -0,0 +1,457 @@
+                    TOFSensorPolicy* policy, TOFSensorID sensor_id) -> bool {
+        if (!_policy) {
+            _policy = policy;
+        }
+
+        _registers = registers;
+
+        // Need to wait when you toggle the init pin.
+        _policy->sleep_ms(DEFAULT_SLEEP_MS);
+        _policy->enable_tof_sensor(sensor_id, true);
+        // Wait for the sensor to start
+        _policy->sleep_ms(DEFAULT_SLEEP_MS * 2);
+
+        // Make sure the sensor is ready
+        if (!set_sensor_ready(sensor_id)) {
+            return false;
+        }
+
+        // Load the image from flash if the sensor is in bootloader mode
+        if (get_sensor_mode(sensor_id) == TOFSensorMode::BOOTLOADER) {
+            if (!handle_bootloader(sensor_id)) {
+                return false;
+            }
+        }
+
+        // Get the custom address this device should have
+        auto address = get_sensor_i2c_address(sensor_id, false);
+        if (!set_sensor_i2c_address(sensor_id, address)) {
+            return false;
+        }
+
+        //// TODO: Configure the spad maps
+        // if (!configure_sensor(registers, sensor_id)) {
+        //    return false;
+        //}
+
+        return true;
+    }
+
+    auto update_enable(const TMF8821RegisterMap& registers,
+                       TOFSensorID sensor_id) -> bool {
+        auto reg = registers.enable;
+        reg.padding_1 = 0;
+        return set_register(reg, sensor_id).has_value();
+    }
+
+    auto write(TOFSensorID sensor_id, uint16_t reg, uint32_t* data,
+               int size = 1) -> std::optional<RegisterSerializedType> {
+        using RT = std::optional<RegisterSerializedType>;
+        auto dev_address = get_sensor_i2c_address(sensor_id);
+        auto [res, _] =
+            _policy->i2c_write(dev_address << 1, reg, (uint8_t*)data, size);
+        if (res != 0) return RT();
+        return RT(res);
+    }
+
+    auto read(TOFSensorID sensor_id, uint32_t reg, int size = 1)
+        -> std::optional<RegisterSerializedType> {
+        using RT = std::optional<RegisterSerializedType>;
+        auto dev_address = get_sensor_i2c_address(sensor_id);
+        auto [res, data] = _policy->i2c_read(dev_address << 1, reg, size);
+        if (res != 0) return RT();
+        auto value = static_cast<uint32_t>(*data.data());
+        return RT(value);
+    }
+
+    /* Check which app (bootloader or measurement ) is running. */
+    auto get_sensor_mode(TOFSensorID sensor_id) -> TOFSensorMode {
+        auto ret = read_register<tmf8821::AppID>(sensor_id);
+        if (!ret.has_value()) {
+            return TOFSensorMode::UNKNOWN;
+        }
+        auto appid = static_cast<tmf8821::AppID>(ret.value()).appid;
+        return static_cast<TOFSensorMode>(appid);
+    }
+
+    auto reset_custom_address() -> void { _custom_address = false; }
+
+    // Gets the sensor i2c address
+    auto get_sensor_i2c_address(TOFSensorID sensor_id, bool default_addr = true)
```

Comment:
I'm a little confused by the logic with having both the `default_addr` param and the built-in `_custom_address` field - why do we need both?

Reply:
Good call, `default_addr` is redundant.